doc/guides/dts/api-usage.rst

```diff
@@ -8,37 +8,60 @@ from C source files. It assumes you're familiar with the concepts in
+devicetree data structure in its binary form. The binary representation is
+parsed it at runtime, for example to load and initialize device drivers.
+
+Zephyr does not work this way because the size of the devicetree binary and
+associated handling code would be too large to fit comfortably on the
+relatively constrained devices Zephyr supports.
+
 .. _dt-node-identifiers:
 
 Node identifiers
 ****************
 
 To get information about a particular devicetree node, you need a *node
 identifier* for it. This is a just a C macro that refers to the node.
-There are four types, each of which is created with a different macro:
-
-Path identifiers
-   These use the node's full path in the devicetree, starting from the
-   root node. They are mostly useful if you happen to know the exact node
-   you're looking for. Create these with :c:func:`DT_PATH()`.
-
-Node label identifiers
-   Unique names which can be given to each node where it is
-   defined. These are often used when the SoC :file:`.dtsi` gives nodes
-   names that match the SoC datasheet, like ``i2c_1``, ``spi_2``, etc.
-   Create these with :c:func:`DT_NODELABEL()`.
-
-Alias identifiers
-   These are created for properties of the special ``/aliases`` node. They are
-   useful for supporting applications (like :ref:`blinky <blinky-sample>`,
-   which uses the ``led0`` alias) that need to refer to *some* device of a
-   particular type ("the board's user LED") but don't care which exact one is
-   used. Create these with :c:func:`DT_ALIAS()`.
-
-Instance identifiers
-   These are used primarily by device drivers, as they're a way to refer to
-   nodes based on their matching compatible. Create these with
-   :c:func:`DT_INST()`, but be careful doing so.
+
+These are the main ways to get a node identifier:
+
+By path
+   Use :c:func:`DT_PATH()` along with the node's full path in the devicetree,
+   starting from the root node. This is mostly useful if you happen to know the
+   exact node you're looking for.
+
+By node label
+   Use :c:func:`DT_NODELABEL()` to get a node identifier from a :ref:`node
+   label <dt-node-labels>`. Node labels are often provided by SoC :file:`.dtsi`
+   to give nodes names that match the SoC datasheet, like ``i2c1``, ``spi2``,
+   etc.
+
+By alias
+   Use :c:func:`DT_ALIAS()` to get a node identifier for a property of the
+   special ``/aliases`` node. This is sometmes done by applications (like
+   :ref:`blinky <blinky-sample>`, which uses the ``led0`` alias) that need to
+   refer to *some* device of a particular type ("the board's user LED") but
+   don't care which one is used.
+
+By instance number
+   This is done primarily by device drivers, as instance numbers are a way to
+   refer to individual nodes based on a matching compatible. Get these with
+   :c:func:`DT_INST()`, but be careful doing so. See below.
+
+By chosen node
+   Use :c:func:`DT_CHOSEN()` to get a node identifier for ``/chosen`` node
+   properties.
+
```

Comment:
should we add parent and child in here?

Reply:
I can if you think we should, but I was hoping to keep the list as small as possible for an introductory guide. There are other cases that return node identifiers -- devicetree/spi.h has at least one. Figured we'd leave those as "advanced" for now.